### PR TITLE
Users/raulga/boost

### PR DESCRIPTION
**Pull request by raulgarciamsft**
Adding queries for boost.org TLS usage

### CODE REVIEW

**Comment by geoffw0** [comment]
I've just created https://github.com/raulgarciamsft/ql/pull/5 to autoformat the QL code in this PR.

File: cpp/ql/test/query-tests/Likely Bugs/Protocols/boostorg/asio/boost_simulation.hpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#define SSL_OP_ALL										 0x80000BFFU
+/* Removed from OpenSSL 1.1.0. Was 0x01000000L */
```
Comment (geoffw0) [inline cpp/ql/test/query-tests/Likely Bugs/Protocols/boostorg/asio/boost_simulation.hpp]: I just want to check, is this a comment you wrote to explain the difference between the versions, or was it copied from a boost header file?

Comment (jbj) [reply cpp/ql/test/query-tests/Likely Bugs/Protocols/boostorg/asio/boost_simulation.hpp]: Perhaps it came from the OpenSSL project? https://github.com/openssl/openssl/blob/e7fb44e7c3f7a37ff83a6b69ba51a738e549bf5c/include/openssl/ssl.h#L462

**Comment by geoffw0** [comment]
Pushed autoformat.

File: cpp/ql/test/query-tests/Likely Bugs/Protocols/boostorg/asio/boost_simulation.hpp (hunk 1 above)
Comment (geoffw0) [reply cpp/ql/test/query-tests/Likely Bugs/Protocols/boostorg/asio/boost_simulation.hpp]: Ah, good spot.  In that case we really shouldn't include it as it's a copy-paste.  Will remove it shortly...

Comment (geoffw0) [reply cpp/ql/test/query-tests/Likely Bugs/Protocols/boostorg/asio/boost_simulation.hpp]: Done.

File: cpp/ql/src/Likely Bugs/Protocols/boostorg/TlsSettingsMisconfiguration.qhelp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+  <p>Using TLS or SSLv23 protool from the boost::asio library, but not disabling deprecated protocols or disabling minimum-recommended protocols.</p>
```
[the remark this answers is not in the document]
Comment (geoffw0) [reply cpp/ql/src/Likely Bugs/Protocols/boostorg/TlsSettingsMisconfiguration.qhelp]: Fixes here: https://github.com/Semmle/ql/pull/2372

**Pull request merged**